agent/tcs/client/client.go

```diff
@@ -241,3 +327,20 @@ func copyTaskMetrics(from []*ecstcs.TaskMetric) []*ecstcs.TaskMetric {
 	copy(to, from)
 	return to
 }
+
+// copyHealthMetadata performs a deep copy of HealthMetadata object
+func copyHealthMetadata(metadata *ecstcs.HealthMetadata, fin bool) *ecstcs.HealthMetadata {
+	return &ecstcs.HealthMetadata{
+		Cluster:           aws.String(*metadata.Cluster),
```

Comment:
I think you can just do `Cluster: metadata.Cluster` here (same for other string fields in this block.

Reply:
No, `metadata.xx` is a pointer, but we want a deep copy here, so that the actual value wasn't changed elsewhere.

Reply:
sorry, missed that. please use aws.StringValue here as well. Avoid doing pointer dereferencing as much as possible.

Reply:
This is still an outstanding item